File: week8/lab_py/bank/bank.py
from customer import Customer

class Bank:
    def __init__(self):
        self.customers = []
        self.add_customers()
    
    def add_customers(self):
        while len(self.customers) < 3:
            c = Customer() 

            if self.check_customer(c.name):
                print("Customer name in use. Please enter a new customer name.")
                continue

            self.customers.append(c)
            print(f"Customer '{c.name}' added successfully.")
    
    def check_customer(self, name):
        for c in self.customers:
            if c.match(name):
                return c
        return None
    
    def read_name(self):
        print("Enter Customer Login Name: ", end="")
        return input().strip()
    
    def customer_login(self):
        c = self.check_customer(self.read_name())
        if c:
            c.use()
        else:
            print("Customer does not exist")

    def show(self):
        customer = self.check_customer(self.read_name())
        if (customer != None):
            print(customer)
        else:
            print("Customer does not exist")

    def add(self):   
        newCustomer = Customer()
        customer = self.check_customer(newCustomer.name)
        if (customer == None): 
            self.customers.append(newCustomer)
        else:
            print("Customer already exists")
    
    def remove(self):
        customer = self.check_customer(self.read_name())
        if (customer != None):
            self.customers.remove(customer)
        else:
            print("Customer does not exist")
    
    def view(self):
        for customer in self.customers:
            print(customer)

File: week9/lab_java/bank/Bank.java
package bank;

import java.util.List;

class Bank {
    private List<Customer> customers;

    public Bank() {
        Config.initialize();
        customers = Config.read();
    }

    private Customer customer(String name){        
        for(Customer c:customers)
            if(c.match(name))
                return c;
        return null;
    }

    private String readName() {
        System.out.print("Enter Customer Name: ");
        return Config.IN.next();
    }

    public void show() {
        Customer customer = customer(readName());
        if (customer != null)
            System.out.println(customer);
        else
            System.out.println("Customer does not exist");
    }
    
    public void view() {
        customers = Config.read();
        customers.forEach(System.out::println);
    }

    public void add(){        
        String name = readName();
        Customer customer = customer(name);
        customers = Config.read();
        if (customer == null) {            
            customers.add(new Customer(name));
            System.out.println("Customer '" + name + "' added successfully.");
            Config.write(customers);
        } else {
            System.out.println("Customer already exists");
        }
    }
    
    public void remove(){
        Customer customer = customer(readName());
        if (customer != null){
            customers.remove(customer);
            System.out.println("Customer '" + customer.getName() + "' removed.");
            Config.write(customers);
        } else
            System.out.println("Customer does not exist");
    }
    
    public void customerLogin() {
        Customer c = customer(readName());
        if (c != null) {
            int index = customers.indexOf(c);
            c.use();
            customers.set(index, c);
            Config.write(customers);
        } else {
            System.out.println("Customer does not exist");
        }
    }
}
